### ISSUE TITLE
WantCaptureMouse  changed?

### ISSUE THREAD

**Issue by sonoro1234** [ocornut/imgui#1789]
When doing a bind against 1.60 the behaviour of WantCaptureMouse changed

this is my callback
```
local function mouse_button_callbak(win, button, action,mods)
			if button == glc.GLFW_MOUSE_BUTTON_1 then
				GL.mouse_but[1] = (action == glc.GLFW_PRESS) and 1 or 0
			end
			imgui.ImGui_ImplGlfwGL3_MouseButtonCallback(win, button, action, mods)
			--dont process mouse if used by imgui
			if imgui.igGetIO().WantCaptureMouse then return end
```
with 1.53 WantCaptureMouse was detected and not now. Has been any relevant change?

**Comment by ocornut** [comment]
No particular change as far as I remember. It's hard to tell how you could have been affected without the fuller context.

**Comment by ocornut** [comment]
You should add code to check the size of structs.. See what I do in `DebugCheckVersionAndDataLayout` + adding also checking IM_OFFSET() of a few selected fields.

**Comment by ocornut** [comment]
@Nukem9 I am not sure what you mean. Do you suggest using `#pragma pack()` (with empty parenthesis) to force-reset the data structure to the compiler initial settings? (surrounding the whole file with push/pop of course) ?

**Comment by Nukem9** [comment]
> @Nukem9 I am not sure what you mean. Do you suggest using #pragma pack() (with empty parenthesis) to force-reset the data structure to the compiler initial settings? (surrounding the whole file with push/pop of course) ?

Yes - that would fix anything like https://github.com/ocornut/imgui/issues/1769 in the future. I'm not sure about fixing the #define problem though (https://github.com/ocornut/imgui/issues/1695).

Personally I wouldn't use `#pragma pack()`, but some number like `#pragma pack(push, 1, 4, 8, 16, ...)` that forces all imgui structs to be the same alignment in every cpp file.

**Comment by sonoro1234** [comment]
My specific problem was not with packing but with a bad definition (now corrected)

**Comment by ocornut** [comment]
@Nukem9 To answer your question, the issue is mostly that it is work to figure out the least intrusive way to do it across most compilers and test it. If you want to do it for imgui.h and imgui_internal.h and it's not too intrusive, I would take a PR. I think Clang needs __attribute__ on every structure :/
